keras/utils/data_utils.py

```diff
@@ -55,23 +56,169 @@ def chunk_read(response, chunk_size=8192, reporthook=None):
     from six.moves.urllib.request import urlretrieve
 
 
+class Archive(object):
+    """ Defines an inteface to extract archive files like tar and zip files.
+    """
+    signature = None
```

Comment:
This type of class property is dangerous (e.g. mutating it in one instance mutates it for all instances...). Do not use it. Rather, set instance properties in `__init__`.